cpp/libs/src/asiopal/TCPClient.cpp

```diff
@@ -123,7 +123,11 @@ void TCPClient::HandleResolveResult(
 		// attempt a connection to each endpoint in the iterator until we connect
 		auto cb = [self = shared_from_this(), callback](const std::error_code & ec, asio::ip::tcp::resolver::iterator endpoints)
 		{
-			callback(self->executor, std::move(self->socket), ec);
+			self->connecting = false;
```

Comment:
There's some duplication here with the code above this. Possibly could extract out into a common method.